### ISSUE TITLE
Thumbs not showing on one of my phones

### ISSUE THREAD

**Issue by HYBRID-BEING** [calvinaquino/LNReader-Android#294]
I'm trying to migrate BakaReader Ex from one of my phones (ASUS Zenfone 2, Android 5.1.1) to another (ASUS Zenfone Selfie, Android 6.0.1). Something seems to be wrong, though - no matter what i try, thumbs don't seem to work on my new phone and all i see is placeholders (sometimes with alt text, sometimes not).

I've tried:
- Moving data with Titanium Backup (TB) to a fresh installation of BakaReader Ex or one restored with TB (1.1.37, yeah, first time i actually tried to migrate was more than 4 years ago).
- Moving _data/com.erakk.lnreader_ manually to a fresh installation or one restored with TB (1.1.37).
- Making novel database and images backups and importing them in a fresh installation or in one restored with TB (1.1.37). Images backup doesn't seem to back up thumbs, actually, only full-res images - trying to make an image backup when **"Process All Images"** is disabled gives me **"Failed to zip files: No entries"** error. Manually copying thumbs (which are stored directly in _/files/project/_ directory) afterwards didn't help.

I've also tried to relink images after every thing i tried, but it didn't help - it always said that 0 images were updated. I took a look in the log files ([here's one](https://github.com/calvinaquino/LNReader-Android/files/5921566/RelinkImageTask_20210123_021240.log)), and i didn't see anything there concerning thumbs.

**Comment by HYBRID-BEING** [comment]
So, i've tried updating my Selfie to LineageOS 15.1 (Android 8.1) - made no difference.

I also checked the fresh installation and noticed that thumbs are placed into the _/files/project/thumbs_ directory (sorted in single-character-named folders by filename's first character), unlike on my Zenfone 2. So i tried batch renaming the files into corresponding format and putting them in corresponding folders, but that didn't help either.

Here's what i have, every thumb is just like that. Covers are okay though, surprisingly.
![Screenshot_BakaReader_EX_20210212-045445](https://user-images.githubusercontent.com/1691231/107714698-889e6e80-6cef-11eb-8435-001066ebde87.png)

**Comment by Nandaka** [comment]
hiya, the apps not actively developed anymore. I'm using https://novelupdates.inverse.cc/ now

**Comment by HYBRID-BEING** [comment]
Yeah, i did notice that was the case, but i've got some novels downloaded that were C&D'd, and wanted to preserve those, at the very least. This seems very weird to me that everything works fine on old phone, yet fails to work on a newer one. Maybe there is some other log i can look at to try to figure this out?

**Comment by HYBRID-BEING** [comment]
Oh boy, i think i actually resolved my conundrum.

I've been juggling files around and it finally hit me that if BakaReader doesn't fix thumbs in the database for me, then i just need to do it myself.
So i got myself DB Browser, opened up _**pages.db**_, found the table of interest (**_novel_books_content_**), extracted it into CSV and instantly noticed that most of the thumbs have paths like _**file:///storage/sdcard0/Android/data/com.erakk.lnreader/files/project/thumb.php_f=High_school_DxD_Volume_19_Cover-Mini.jpg&amp;width=85**_, while few have something like _**file:///storage/emulated/0/Android/data/com.erakk.lnreader/files/project/thumb.php_f=C3_01-Cover.jpg&amp;width=82**_. Turns out some of the thumbs actually worked on my Selfie! But latter work on the Zenfone 2 as well - i've checked directories and it seems that, unlike on Selfie, **_/sdcard0/_** and **_/emulated/0/_** were symlinked.
A few regexp replaces later, i have a CSV with paths changed to _**/emulated/0/**_ (also, changed weird php names to _**/thumbs/N/Name-XXpx.ext**_, while at it), imported it into the db file in place of original table (don't forget to set attributes as they originally were, or BakaReader will slow down to a crawl!), and, voila, thumbs are present once again.

Though, i think there might be some weird files left like **_file:///storage/emulated/0/Android/data/com.erakk.lnreader/files/project/thumbs/C/CtG_v02_%3F-84px.jpeg_**. The actual file was saved as **_CtG_v02__-84px.jpeg_**, so BakaReader doesn't pick it up.
